gossip/src/cluster_info.rs

```diff
@@ -929,13 +929,21 @@ impl ClusterInfo {
         &self,
         transaction: &Transaction,
         tpu: Option<SocketAddr>,
+        connection_cache: &Arc<ConnectionCache>,
     ) -> Result<(), GossipError> {
         let tpu = tpu
             .map(Ok)
-            .unwrap_or_else(|| self.my_contact_info().tpu(contact_info::Protocol::UDP))?;
+            .unwrap_or_else(|| self.my_contact_info().tpu(connection_cache.protocol()))?;
         let buf = serialize(transaction)?;
-        self.socket.send_to(&buf, tpu)?;
-        Ok(())
+        let client = connection_cache.get_connection(&tpu);
+        let result = client.send_data_async(buf);
```

Comment:
If we are not using `ClusterInfo.socket` anymore, then I don't think this code needs to live in gossip.
I think you can just move this function to `core/src/voting_service.rs`

Reply:
Moved to voting_service.rs